Работа с тестовыми данными

denzel-commits opened the pull request:
- [x] В репозитории отсутствуют лишние файлы
- [x] Соблюдается минимальный кодстайл (встроенный в PyCharm) 

> проверено flake8

- [x] В личном кабинете или репозитории приложен файл result.json с итоговым результатом. 
> Путь к файлу: /data/output_files/result.json

- [x] Исходные файлы копировать не нужно.

denzel-commits committed:
Feature: read books and users files

denzel-commits committed:
Feature: implement 'distribute_books' function, refactor utils function names

denzel-commits committed:
Refactor: rename vars

denzel-commits committed:
Refactor: remove blank line at end of file

denzel-commits committed:
Tests: add tests with pydantic schema

denzel-commits committed:
update gitignore

denzel-commits committed:
Doc: add docstring to tests

denzel-commits committed:
Doc: add README.md

denzel-commits committed:
Refactor: separate pydantic schemas, add expected model in test_result_structure parametrize

denzel-commits committed:
Fix: import schema name

e7506c commented on main.py:
```diff
@@ -1,16 +1,37 @@
+            "genre": book["Genre"],
+        })
+
+    result_dict = []
```
можно без цикла через компрехеншн (рассказывал на занятии по финкциональному программированию)

```
result_dict = [{
    "name": user["name"],
    "gender": user["gender"],
    "address": user["address"],
    "age": user["age"],
    "books": collected_books[user["_id"]],
} for user in users]
```

e7506c commented on main.py:
```diff
@@ -1,16 +1,37 @@
+            "books": collected_books[user["_id"]],
+        })
+
+    json_dump(result_dict, result_file)
```
тут у вас нарушается single responsibility принцип. лучше разделить функцию которая распределяет книги и которая сохраняет в файл. с точки зрения архитектуры так будет гибче и проще писать тесты.

e7506c commented on src/utils.py:
```diff
@@ -0,0 +1,40 @@
+    if os.path.exists(filename):
+        with open(filename, "r", newline="") as csvfile:
+            return list(csv.DictReader(csvfile))
+    else:
```
тут не нужен else. можно просто:

```
    if os.path.exists(filename):
        with open(filename, "r", newline="") as csvfile:
            return list(csv.DictReader(csvfile))
    raise FileNotFoundError(
            f'Файл {filename} не найден,\n'
            'пожалуйста, предоставьте файл по указанному пути')
```

e7506c commented on src/utils.py:
```diff
@@ -0,0 +1,40 @@
+    if os.path.exists(filename):
+        with open(filename, "r") as f:
+            return json.load(f)
+    else:
```
аналогично

e7506c commented on tests/test_result.py:
```diff
@@ -0,0 +1,23 @@
+
+
+@pytest.mark.parametrize("result_file_path, model", [(OUTPUT_FILE_PATH+"result.json", User)])
+def test_result_structure(result_file_path, model, get_json):
```
тесты - это очень хорошая идея

denzel-commits committed:
Refactor:
change for user in users to comprehension
update utils function with guard expressions

denzel-commits committed:
Refactor: optimize distribute_books function

denzel-commits committed:
Refactor: change path to os.path.join, fix [books] key to store books
